### PR TITLE
Fix popover fade-out animation, add darker variation of the popover background

### PR DESCRIPTION
**Pull request by kdzwinel**
Tested on Chrome, Firefox, Safari, Edge and IE11.

## Fade-out animation fix

Popover fade-out animation wasn't working because 'visibility: hidden' was immediately applied after 'is-shown' class was removed hiding the whole thing instantaneously.

*BEFORE* Gif from production (animation slowed down to 1/4 of regular speed):

modal-visibility-broken mov

The fix was to also animate 'visibility'. Also, z-index has to be 300 through the whole animation otherwise search bar will show over the modal background.

*AFTER* (animation slowed down to 1/4 of regular speed):

modal-visibility-fixed mov

## Darker background variation

In the manual location project we need a better contrast between the background and the map shown in the popup. See project designs for more info.

### CODE REVIEW

File: objects/_modal-popover.scss [hunk 1]
```diff
@@ -94,3 +94,9 @@ $popover-shadow: 0 0 15px -3px rgba(0,0,0,0.35);
     }
 }
 
+.modal--popover--dark {
```
Comment (kdzwinel) [inline objects/_modal-popover.scss]: we already have variation named `--dk`, so TBH I'm not sure about `--dark` 🤔

Comment (bsstoner) [reply objects/_modal-popover.scss]: yea, the `--dk` one isn't very dark, I think it was initially and then they changed it. Calling your new one one dark seems right, maybe we rename `--dk`? `--gray`? `--medium`?

Comment (bsstoner) [reply objects/_modal-popover.scss]: I merged it, we can address the `--dk` naming separately.

Comment (kdzwinel) [reply objects/_modal-popover.scss]: Yeah, good call. I'll create a PR renaming `--dk` to `--gray`

**Pull request merged**